### ISSUE TITLE
Write protection should not vary between debug and prod mode

### ISSUE THREAD

**Issue by alunny** [flightjs/flight#376]
Discussed briefly with @phuu and mentioned to @lawnsea.

Right now the `canWriteProtect` function in utils behaves different in debug mode from production/normal mode:
https://github.com/flightjs/flight/blob/master/lib/utils.js#L12

If `debug.enabled` is true, mixins cannot overwrite functions from earlier mixins. Otherwise, later mixins clobber functions from earlier mixins.

This can lead to nasty bugs where a component will behave differently when debugging from production mode.

I'd like to remove the `setWritability` calls entirely from `compose.mixin`, since that's how the library behaves in production. The alternative way to eliminate the discrepancy would be to always write protect, which is much slower.

Thoughts?

**Comment by angus-c** [comment]
If clobbering is detected in debug doesn't it throw an error? If it does than I think that's fine. If it doesn't then it should.

**Comment by alunny** [comment]
Nope, writing to a non-writable property does not throw an error. It just fails silently.

```js
> foo = {}
{}
> foo.bar = 'hello'
'hello'
> Object.defineProperty(foo, 'bar', { writable: false })
{ bar: 'hello' }
> foo.bar = 'goodbye'
'goodbye'
> foo.bar
'hello'
```

**Comment by angus-c** [comment]
We should make it error (it used to). Unless no-one cares about clobbering (but they probably should)

**Comment by alunny** [comment]
It looks like this behavior's been the same since the first open source commit, at least.

**Comment by angus-c** [comment]
yeah you right, idk.

**Comment by giuseppeg** [comment]
weird it should throw in strict mode

**Comment by alunny** [comment]
(long delayed response).

> weird it should throw in strict mode

Yeah my mistake, I was running the node repl in sloppy mode.

That said, flight cannot control whether the code that uses it runs in strict mode or sloppy mode. So I think the problem is still valid.
